ql/src/java/org/apache/hadoop/hive/ql/exec/OrcFileMergeOperator.java

```diff
@@ -154,6 +154,9 @@ private void processKeyValuePairs(Object key, Object value)
 
       // next file in the path
       if (!k.getInputPath().equals(prevPath)) {
+        if (reader != null) {
```

Comment:
do we need to do this on line 111 as well? or at that point it's guaranteed to be the first reader instantiation?

Reply:
I think the line 111 is for the first time in the loop - we do not have a previous path/reader at that time